src/Summary/SummaryManager.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+namespace AppBundle\Summary;
+
+use AppBundle\Entity\Adherent;
+use AppBundle\Entity\Summary;
+use AppBundle\Repository\SummaryRepository;
+
+class SummaryManager
+{
+    private $factory;
+    private $repository;
+
+    public function __construct(SummaryFactory $factory, SummaryRepository $repository)
+    {
+        $this->factory = $factory;
+        $this->repository = $repository;
+    }
+
+    public function getForAdherent(Adherent $adherent): Summary
+    {
+        if ($summary = $this->repository->findOneForAdherent($adherent)) {
+            return $summary;
+        }
+
+        return $this->factory->createFromAdherent($adherent);
```

Comment:
You will need to be careful to persist this properly as it will not be managed by Doctrine in this case but will be in the other case.